service/task.go

```diff
@@ -889,3 +886,84 @@ func (uis *UIServer) testLog(w http.ResponseWriter, r *http.Request) {
 		uis.render.WriteResponse(w, http.StatusOK, data, "base", template)
 	}
 }
+
+func readBuildloggerToSlice(ctx context.Context, taskID string, r io.ReadCloser) []apimodels.BuildloggerLogLine {
+	lines := []apimodels.BuildloggerLogLine{}
+	lineChan := make(chan apimodels.BuildloggerLogLine, 1024)
+	go readBuildloggerToChan(ctx, taskID, r, lineChan)
+
+	for {
+		line, more := <-lineChan
```

Comment:
you might do this in a select statment with a default

Reply:
what would the default do?